services_and_applications/slate/use_cases/gitlab_runner.rst

```diff
@@ -0,0 +1,128 @@
+modify the requests and limits as needed.
+
+This helm chart was forked from the the upstream GitLab Runner helm chart. As such, nearly all of the parameters documented in the
+upstream chart could be provided here as well.
+
+To verify that the GitLab runner was properly deployed, one could explore the deployed resources from the Topology view from the
+Slate web console. Or, from the command line:
+
+.. code-block:: bash
+
+   $ helm ls
+   NAME         	NAMESPACE    	REVISION	UPDATED                               	STATUS  	CHART              	APP VERSION
+   gitlab-runner	myproject   	1       	2021-09-15 14:10:15.12187869 +0000 UTC	deployed	gitlab-runner-1.0.0	14.2.0
+   $ oc get pods
+   NAME                                          READY   STATUS    RESTARTS   AGE
+   gitlab-runner-gitlab-runner-687486d94-lpmhs   1/1     Running   0          107s
+
+Once the pod has a Status of "Running", navigate to the GitLab web interface and ensure that the runner has registered to either the group or repository per the token given, and that it is listed as available.
+
+Finally, if batch scheduler integration was enabled, one can verify functionality in the pod with:
+
+.. code-block:: bash
+
+   $ oc exec -it pods/gitlab-runner-gitlab-runner-687486d94-lpmhs -- bash
+   bash-5.0$ squeue
+             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+              9951 ...
+
+Removing a GitLab Runner
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To remove a namespace installed GitLab runner, login to the cluster, change to the appropriate contact, and use the helm command line to
+uninstall the instance. This would look similar to:
+
+.. code-block:: bash
+
+   $ oc login
+   Authentication required for https://api.marble.ccs.ornl.gov:443 (openshift)
+   Username: millerjl
+   Password:
+   Login successful.
+
+   Using project "myproject".
+   $ helm ls
+   NAME         	NAMESPACE    	REVISION	UPDATED                                	STATUS  	CHART              	APP VERSION
+   gitlab-runner	myproject   	1       	2021-09-13 16:52:27.938127657 +0000 UTC	deployed	gitlab-runner-1.0.0	14.2.0
+   $ helm uninstall gitlab-runner
+   release "gitlab-runner" uninstalled
+
+Verify that the runner has been unregistered from the GitLab project (GitLab->Settings->CI/CD->Runners) and check that all the pods have
+been removed from the namespace:
+
+.. code-block:: bash
+
+   $ oc get pods
+   NAME                                          READY   STATUS        RESTARTS   AGE
+   gitlab-runner-gitlab-runner-687486d94-lkcbk   0/1     Terminating   0          24h
+
+Notice, in this instance, a pod remains after the runner has been registered and the termination grace
+period (34600 seconds by default) has passed. To remove the pod in this instance, a force deletion may
+be needed:
+
+.. code-block:: bash
+
+   $ oc delete pod gitlab-runner-gitlab-runner-687486d94-lkcbk --force
```

Comment:
Forcing a delete can have negative consequences and probably shouldn't be mentioned in user-facing documentation without more caveats since what we are doing is telling the Kube API to forget about the pod but it will remain running on the host under the CRI until it is killed.

If the pod isn't shutting down like it should in a reasonable amount of time it sounds like a bug of some sort. The grace period is the maximum that the pod will be allowed to live before getting a SIGKILL signal